modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -888,14 +903,6 @@ private CompletableFuture<Void> awaitFragmentInitialisationAndClose() {
 
                 cancelFuts.add(
                         CompletableFuture.allOf(entry.getValue().toArray(new CompletableFuture[0]))
-                                .handle((none2, t) -> {
-                                    // Some fragments may be completed exceptionally, and that is fine.
-                                    // Here we need to make sure that all query initialisation requests
-                                    // have been processed before sending a cancellation request. That's
-                                    // why we just ignore any exception.
-
-                                    return null;
-                                })
```

Comment:
why did you remove this?

Reply:
))) as i can observe through history initially this section process exception handling and after just degenerate into empty block. I agree To fill the issue for this case if you ok.

Reply:
have you read comment in this empty block?

Reply:
of course, but still have no clue, i found that no one test will touch this place with not null **t**

Reply:
it's not news that we don't have comprehensive test coverage.

In the very beginning, the comment states "Some fragments may be completed exceptionally". Therefore, we need to find out whether this invariant is conserved. There are only 5 usages `remoteFragmentInitCompletion`, so I believe it's not a rocket science to take a look at every usage. As far as I can see, `DistributedQueryManager#onNodeLeft` completes init future with exception. 

Feel free to add such test to ExecutionServiceImplTest. If it will pass without mentioned `handle` block, I'm ok to remove it